src/stores/user.ts

```diff
@@ -1,38 +1,36 @@
 import { create } from "zustand";
 import { User } from "firebase/auth";
+import { Profile } from "@/types";
+import { profileIsPremium } from "@/utils";
 
 interface IUseUserStore {
   user: User | null;
-  user_id: string;
-  email: string;
-  tier: string;
+  profile: Profile | null;
+  isLoading: Boolean;
   setUser: (user: User | null) => void;
-  setUserId: (id: string) => void;
-  setEmail: (email: string) => void;
-  setTier: (tier: string) => void;
+  setProfile: (profile: Profile | null) => void;
+  setIsLoading: (status: boolean) => void;
+  getPremiumStatus: () => boolean;
 }
 
-const useUserStore = create<IUseUserStore>((set) => ({
+const useUserStore = create<IUseUserStore>((set, get) => ({
```

Comment:
If we are exporting default, we should rename the files according to the variable.  i.e. `useUserStore.ts` instead of `user.ts`.

Same for other stores.